spec/requests/parents_request_spec.rb

```diff
@@ -0,0 +1,270 @@
+require 'rails_helper'
+
+RSpec.describe "Admin::Parents", type: :request do
+  let(:parent) { create(:parent) }
+  let(:parents) { create(:parent, 3) }
+  let!(:admin) { create(:user, :admin) }
+  let!(:user) { create(:user) }
+  let!(:not_user) { nil }
+  let!(:invalid_id) { '998' }
+  let(:not_parent) { nil }
+  let(:new_user) {create(:user)}
+
+  describe 'GET #index' do
+    context 'with logged-in admin' do
+      before do
```

Comment:
move this block to describe section since it's similar in each context
define in each context some user variable and use it here, for example:
```
before do
  sign_in current_user
  ...
```
```
let(:current_user) { admin }
```

Reply:
refactored.